MyFridge/MyFridge/Refrigerator.cs

```diff
@@ -0,0 +1,235 @@
+                {
+                    shelf.InsertingItem(item);
+                    return;
+                }
+                throw new Exception("There is not enough space");
+
+            }
+  
+        }
+
+
+        public Item TookItem(int itemId)
+        {
+            Item empty;
+            foreach (var shelf in Shelves)
+            {
+                empty = shelf.TookItem(itemId);
+                if (empty != null) return empty;    
+
+            }
+             throw new Exception("No item if such ID number");
+           
+            
+        }
+
+        public void Cleaning()
+        {
+            foreach (Shelf shelf in Shelves)
+            {
+                shelf.Cleaning();
+            }
+        }
+
+        public List<Item> WantToEat(Kosher kosher, KindItem kind)
+        {
+            List<Item> items = new();
+            foreach (Shelf shelf in Shelves)
+            {
+                if (shelf.WantToEat(kosher,kind)!=null)
+                {
+                    items.AddRange(shelf.WantToEat(kosher, kind));
+                }
+            }
+            if (items.Count == 0) throw new Exception("Unfortunately there is currently no suitable food. ");
+            return items;
+        }
+        public List<Item> SortOfExpiryDate()
+        {
+            List<Item> items = new List<Item>();
+            foreach (var shelf in Shelves)
+            {
+                items.AddRange(shelf.Items);
+            }
+            items.Sort((p1, p2) => p1.ExpiryDate.CompareTo(p2.ExpiryDate));
+            
+            return items;
+
+        }
+        public List<Shelf> SortByAvailability()
+        {
+            List<Shelf> shelves= new();
+            shelves.AddRange(Shelves);
+            shelves.Sort((p1, p2) => p1.SpaceAvailable.CompareTo(p2.SpaceAvailable));
+            return shelves;
+
+        }
+        public List<Item> DairProducts(ref int freeSpace)
+        {
+        List <Item> items = new();
+            foreach (var shelf in Shelves)
+            {
+                foreach (var item in shelf.Items)
+                {
+                    if (item.Kosher == Kosher.Dairy && item.ExpiryDate.AddDays(-3) <= DateTime.Today)
+                    {
+                        freeSpace+=item.Space;
+                        items.Add(item);
+                    }
+                }
+            }
+            return items;
+        }
+        public void GoingShopping()
+        {
+            if (this.GetFreeSpace() >= 29)
+            {
+                Console.WriteLine("go to shopping");
+                return;
+            }
+            this.Cleaning();
+            if (this.GetFreeSpace() >= 29)
+            {
+                Console.WriteLine("go to shopping");
+                return;
+            }
+            var freeSpace=this.GetFreeSpace();
+            List<Item> items = new();
+            items.AddRange(this.DairProducts(ref freeSpace));
```

Comment:
cod readability/structure:
reading this line it's hard to understand what's happening. The name indicates it's dealing with Dairy, but it hides the expiry check. Either add this to method name or add the days as a parameter (DairyProducts(3)).

also - a method should do one thing. either it's retrieveing the dairy products or it's calculating the new freeSpace.
I suggest to separate to 2 steps: 1 - retrieve products. 2 - calc the space of items from (1)
this is maybe more lines, but easier to understand and debug :)